chunky/src/java/se/llbit/math/PackedOctree.java

```diff
@@ -692,6 +360,13 @@ public static PackedOctree load(DataInputStream in) throws IOException {
     return tree;
   }
 
+  public static PackedOctree loadWithNodeCount(long nodeCount, DataInputStream in) throws IOException {
```

Comment:
Is this an optimization because it can pre-allocate the array?

Reply:
Yes and we could exit early if there is too much nodes.
I can also imagine this being useful for a new implementation that is read only (can only be constructed from an existing and finalized octree)